main.lua

```diff
@@ -739,12 +734,18 @@ local function parse_path(raw_path)
   if path == 'api' then
     path = path .. "/" .. path_components()
   end
-  local param1 = path_components() or ""
-  local param2 = path_components() or ""
+  if path == 'api/loadfile' then
+    -- The raw path itself might contain '/' which thus needs to be passed as a base64-encoded string.
+    -- The base64-encoded path might itself contain '/', handling that by reversing the path for parsing.
+    local rev_path = string.reverse(raw_path)
+    param1 = dec64(string.reverse(string.sub(rev_path, string.find(rev_path, '/') + 1, string.find(rev_path, string.reverse('api/loadfile')) - 2))) or ""
+    param2 = string.reverse(string.sub(rev_path, 1, string.find(rev_path, '/') - 1)) or ""
+  else
+    param1 = path_components() or ""
+    param2 = path_components() or ""
+  end
```

Comment:
The implementation as is supports url-encoded paths so there is no need to have the paths b64 encoded. Additionally this would result in a breaking change.